lib/dl_connector_bundle_chs3/dl_connector_bundle_chs3_tests/db/conftest.py

```diff
@@ -1,7 +1,23 @@
+from typing import (
+    Any,
+    Optional,
+)
+
+import pytest
+
 from dl_api_lib_testing.initialization import initialize_api_lib_test
 
+from dl_connector_bundle_chs3.chs3_base.core.us_connection import BaseFileS3Connection
 from dl_connector_bundle_chs3_tests.db.config import API_TEST_CONFIG
 
 
 def pytest_configure(config):  # noqa
     initialize_api_lib_test(pytest_config=config, api_test_config=API_TEST_CONFIG)
+
+
+@pytest.fixture(autouse=True)
+def patch_get_full_s3_filename(monkeypatch: pytest.MonkeyPatch) -> None:
+    def _patched(self: Any, s3_filename_suffix: Optional[str]) -> Optional[str]:  # type: ignore
+        return s3_filename_suffix
+
+    monkeypatch.setattr(BaseFileS3Connection, "get_full_s3_filename", _patched)
```

Comment:
Can you please explain why this is necessary?

Reply:
This is the easiest way to keep our current fixtures flow for file-connections in api & core tests (without usage of a real file-uploader). The problem is that the s3 file and the data source description are prepared before connection is saved, and we don't know the future connection id (to combine the filename).

Reply:
Ok, it's just that we had multiple issues in the past with tests after we forget to update the monkeypatched version of some method

Can it be done like this?
https://github.com/datalens-tech/datalens-backend/blob/ad03ba8e11ff3eaba66a5065530e1df173c54772/lib/dl_connector_bundle_chs3/dl_connector_bundle_chs3_tests/db/base/api/base.py#L88-L99